src/main/java/me/jellysquid/mods/sodium/mixin/features/fast_leaf_culling/MixinLeavesBlock.java

```diff
@@ -0,0 +1,29 @@
+package me.jellysquid.mods.sodium.mixin.features.fast_leaf_culling;
+
+import me.jellysquid.mods.sodium.client.SodiumClientMod;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.LeavesBlock;
+import net.minecraft.block.Material;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.option.GraphicsMode;
+import net.minecraft.util.math.Direction;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(LeavesBlock.class)
+public class MixinLeavesBlock extends Block {
+    public MixinLeavesBlock() {
+        super(Settings.of(Material.AIR));
+        throw new AssertionError("Mixin constructor called!");
+    }
+
+    @Override
+    public boolean isSideInvisible(BlockState state, BlockState stateFrom, Direction direction) {
+        if (!(SodiumClientMod.options().quality.leavesQuality.isFancy(MinecraftClient.getInstance().options.graphicsMode))) {
```

Comment:
Minor nitpick: We shouldn't use an inverted if-statement if we're just taking one of two branches... It makes the code easier to read if the logic is instead "If fancy leaves enabled, use default culling logic".